src/PetProject.Framework.Kafka/Logging/GenericKafkaLog.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace PetProjects.Framework.Kafka.Logging
+{
+    using Microsoft.Extensions.Logging;
+
+    /// <summary>
+    /// Wrapper to segregate Kafka logs in kibana.
+    /// </summary>
+    public class GenericKafkaLog
+    {
+        private readonly ILogger logger;
+
+        public GenericKafkaLog(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        public void KafkaLogWarning(string message, params object[] args)
```

Comment:
Hmm this strategy works for our use case as we can filter the kafka specific logs in kibana but the messages will be logged a bit awkwardly. We could try doing it like this: 

```
(logger.BeginScope("LogType: {logType}", LogTypes.Kafka)) 
{ 
    logger.LogWarning(message, args)
}
```

If this approach works, then we could even create an extension method for ILogger in framework-kafka to wrap this BeginScope call so that we have the logtype property in every log produced by the framework.

Reply:
With this approach we would keep the log Message the same and we would have the additional property logType in the logevent so that we could filter it out in kibana.

Reply:
👍 I will change it, I had some doubts about the approach I chose because it was awkward, as you said.